From fb2ba4cc8a2d82962434538ba981b459f34962a8 Mon Sep 17 00:00:00 2001
From: slipsnip <eschmichael@yandex.com>
Date: Fri, 8 May 2020 23:43:53 -0400
Subject: [PATCH] setup multiple databases and inject heroku postgres default

---
 tarot_juicer/settings.py | 17 ++++++++++++++++-
 1 file changed, 16 insertions(+), 1 deletion(-)

diff --git a/tarot_juicer/settings.py b/tarot_juicer/settings.py
index 4b7bbbe..8e74790 100644
--- a/tarot_juicer/settings.py
+++ b/tarot_juicer/settings.py
@@ -86,12 +86,27 @@ WSGI_APPLICATION = 'tarot_juicer.wsgi.application'
 # https://docs.djangoproject.com/en/2.2/ref/settings/#databases
 
 DATABASES = {
+    # default will be updated / replaced  by db_from_env (heroku)
     'default': {
+        'ENGINE': 'django.db.backends.postgresql',
+    },
+    # my local postgres database for testing script automation
+    'local': {
+        'ENGINE': 'django.db.backends.postgresql',
+        'USER': 'wurst',
+        'NAME': 'wurst',
+        'PASSWORD': 'tarotdb',
+    },
+    # the sqlite database, later will switch between
+    'sqlite': {
         'ENGINE': 'django.db.backends.sqlite3',
         'NAME': os.path.join(BASE_DIR, 'db.sqlite3'),
-    }
+    },
 }
 
+# HEROKU - replace / update default database with heroku postgresql 
+db_from_env = dj_database_url.config(conn_max_age=600)
+DATABASES['default'].update(db_from_env)
 
 # Password validation
 # https://docs.djangoproject.com/en/2.2/ref/settings/#auth-password-validators
-- 
2.26.2

